File: src/+utils/matrix_helpers.m
classdef matrix_helpers
    methods (Static)
        function transpose_matrix = TransposeMatrix3d(Phi, Theta)
            transpose_matrix = [cos(Phi').*abs(sin(Theta')); sin(Phi').*abs(sin(Theta')); cos(Theta')];
        end
        function transpose_matrix = TransposeMatrix2d(Phi)
            transpose_matrix = [cos(Phi'); sin(Phi')];
        end
    end
end

File: python/src/utils/matrix_helpers.py
import numpy as np
import math

def transpose_matrix3d(phi, theta):
    return np.array([[math.cos(np.transpose(phi))*abs(math.sin(np.transpose(theta)))], [math.sin(np.transpose(phi))*abs(math.sin(np.transpose(theta)))], [math.cos(np.transpose(theta))]])

def transpose_matrix2d(phi):
    return np.array([[math.cos(np.transpose(phi))], [math.sin(np.transpose(phi))]])
